### PR TITLE
added google sign in

### PR DESCRIPTION
**Pull request by SGR007**
Feature https://github.com/sitamadex11/CovidHelp/issues/19

### CODE REVIEW

**Comment by HeySitam** [comment]
Please give me some time. I will check it soon.

**Comment by HeySitam** [comment]
hey 
```
               val inputMethodManager =
                    getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                inputMethodManager.hideSoftInputFromWindow(viewNew.windowToken, 0)
                viewNew.clearFocus()
                return consumed
            }
        }
        return super.dispatchTouchEvent(ev)
```
here the viewNew variable is giving a unresolved reference error.

**Comment by HeySitam** [comment]
please check it once

**Comment by SGR007** [comment]
> 
> 
> hey
> 
> ```
>                val inputMethodManager =
>                     getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
>                 inputMethodManager.hideSoftInputFromWindow(viewNew.windowToken, 0)
>                 viewNew.clearFocus()
>                 return consumed
>             }
>         }
>         return super.dispatchTouchEvent(ev)
> ```
> 
> here the viewNew variable is giving a unresolved reference error.

i didn't added this code, still i'll check it

**Comment by HeySitam** [comment]
Yeah please check it once.

**Comment by cdhiraj40** [comment]
hey, @SGR007 seems like you have changed the code from #84, maybe try to copy the code which you did in some text files and then ``` git stash```, and just paste it again might help.

**Comment by SGR007** [comment]
@sitamadex11 please check, i have updated the code

File: app/src/main/java/com/sitamadex11/CovidHelp/activity/LoginActivity.kt
Comment (cdhiraj40) [inline app/src/main/java/com/sitamadex11/CovidHelp/activity/LoginActivity.kt:360]: why are you changing this code when you are doing ```google sing-in auth```

**Review by cdhiraj40 — CHANGES_REQUESTED** [review]
@SGR007 Please Don't touch the code which is not part of your PR as it can break the codebase, I would suggest you just add the code which is actually mentioned by your PR, and after that, you can squash the commits which are not changing anything(when you will commit the change --> adding the code which was already there those will be the code which isn't changing anything) you can squash those commits. if you need any help with this do tell me :)
also do remove your PR from the draft after your changes so the maintainer can merge it.

File: app/src/main/java/com/sitamadex11/CovidHelp/activity/LoginActivity.kt
Comment (SGR007) [reply app/src/main/java/com/sitamadex11/CovidHelp/activity/LoginActivity.kt:360]: The code was not merged properly by sitamadex11 into develop, that was the only issue, the only change i done is removing the comment

```kt
    // this  lets keyboard close when clicked in background

    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_DOWN) {
            val v = currentFocus
            if (v is EditText) {
                val outRect = Rect()
                v.getGlobalVisibleRect(outRect)
                if (!outRect.contains(event.rawX.toInt(), event.rawY.toInt())) {
                    v.clearFocus()
                    val imm = getSystemService<InputMethodManager>()?.hideSoftInputFromWindow(
                        v.windowToken,
                        0
                    )
/*
    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
        if (ev.action == MotionEvent.ACTION_UP) {
            val view = currentFocus
            if (view != null) {
                val consumed = super.dispatchTouchEvent(ev)
                val viewTmp = currentFocus
                val viewNew: View = viewTmp ?: view
                if (viewNew == view) {
                    val rect = Rect()
                    val coordinates = IntArray(2)
                    view.getLocationOnScreen(coordinates)
                    rect[coordinates[0], coordinates[1], coordinates[0] + view.width] =
                        coordinates[1] + view.height
                    val x = ev.x.toInt()
                    val y = ev.y.toInt()
                    if (rect.contains(x, y)) {
                        return consumed
                    }
                } else {
                    return consumed
*/
                }
                val inputMethodManager =
                    getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                inputMethodManager.hideSoftInputFromWindow(viewNew.windowToken, 0)
                viewNew.clearFocus()
                return consumed
            }
        }
        return super.dispatchTouchEvent(ev)
    }
```

it was like this

Comment (HeySitam) [reply app/src/main/java/com/sitamadex11/CovidHelp/activity/LoginActivity.kt:360]: Actually, there was a merging conflict. That's why I have resolved it.

**Pull request merged**
